### ISSUE TITLE
MinGW: Redefinition of struct timespec

### ISSUE THREAD

**Issue by TheComet93** [urho3d/Urho3D#1444]
I tried compiling Urho3D on Windows 7, and I ran into a compilation error on this line: https://github.com/urho3d/Urho3D/blob/master/Source/ThirdParty/Civetweb/src/civetweb.c#L329

I was able to fix it by changing it to this:
```
// Urho3D: fix VC15 compilation issue
#if defined(_MSC_VER) && (_MSC_VER >= 1900) 
#define _TIMESPEC_DEFINED
#endif
#if defined(__MINGW32__)
#define _TIMESPEC_DEFINED
#endif
#ifndef _TIMESPEC_DEFINED
struct timespec {
	time_t tv_sec; /* seconds */
	long tv_nsec;  /* nanoseconds */
};
#endif
```

**Comment by cadaver** [comment]
Interesting, I just looked at civetweb's own CMake / makefiles to see if we're missing a configuration test, but didn't find anything related. In any case this looks like a valid fix.

**Comment by weitjong** [comment]
Interesting indeed. We do not have any problems before with our MinGW CI builds or any reports from other MinGW users. A quick check on my MinGW-W64 installation shows that it consistently define the ```_TIMESPEC_DEFINED``` each time it declares the ```struct timespec```.  So, I setup this quick test below.

```
// Urho3D: fixes #1444
#if defined(__MINGW32__)
#ifdef _TIMESPEC_DEFINED
#error Should not be here!
#endif
#define _TIMESPEC_DEFINED
#endif
```

The compiler hit the error pragma which indicates that the ```_TIMESPEC_DEFINED``` is now defined twice, at least for my case.

**Comment by TheComet93** [comment]
This is indeed very strange. It appears my version of mingw32-gcc doesn't define _TIMESPEC_DEFINED even though timespec is declared.

## Test 1
The following code fails to compile.
```cpp
#include <time.h>

#ifndef _TIMESPEC_DEFINED
struct timespec {
    time_t tv_sec; /* seconds */
    long tv_nsec;  /* nanoseconds */
};
#endif

int main() {}
```

Output:
```
$ g++ -W -o test main.cpp
main.cpp:4:8: error: redefinition of 'struct timespec'
 struct timespec {
        ^
In file included from c:\mingw\include\time.h:53:0,
                 from main.cpp:1:
c:\mingw\include\parts\time.h:105:8: error: previous definition of 'struct timespec'
 struct timespec
        ^
```

## Test 2
This compiles fine:
```cpp
#include <time.h>

// Urho3D: fixes #1444
#if defined(__MINGW32__)
#ifdef _TIMESPEC_DEFINED
#error Should not be here!
#endif
#define _TIMESPEC_DEFINED
#endif

int main() {}
```

Output:
```
$ g++ -W -o test main.cpp
./test.exe
```

## Version info
I'm running Windows 7 Ultimate 64-bit in a VM. I used the "MinGW Installation Manager" to install ```mingw32-gcc 4.9.3-1```.

```
$ g++ --version
g++.exe (GCC) 4.9.3
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.```

**Comment by weitjong** [comment]
I guess it is not important now after the latest patch with CMake auto-detection.
